File: pos_tagging.js
class Grammar {
  constructor() {
    this.words = '';
    this.sentences = [];
    this.words_types_list = [];
  }
  parse(text) {
    this.words = text.split(' ');
    let current_sentence = '';
    for (let word of this.words) {
      current_sentence += word + ' ';
      for (let pun of ['.', '!', '?']) {
        if (word.endsWith(pun)) {
          this.sentences.push(current_sentence.trim());
          current_sentence = '';
        }
      }
    }
    if (current_sentence.length > 0) {
      this.sentences.push(current_sentence.trim());
    }
    for (let sentence in this.sentences) {
      for (let word in sentence.split(' ')) {
        this.words_types_list.push(this.getWordType(word));
      }
    }
    return this.words_types_list;
  }
  getWordType(word) {
    const types = new Map();
    const request_dic = new XMLHttpRequest();
    word = word.toLowerCase();
    for (let pun of ['.', '!', '?']) {
      if (word.endsWith(pun)) {
        word.slice(-1);
      }
    }
    request_dic.onload = () => {
      let dictionary = JSON.parse(request_dic.responseText);
    }
    request_dic.open('GET', 'WordNet.json');
    request_dic.send();
    return types.entries();
  }
}
const grammar = new Grammar();
console.log(grammar.parse('I am Umar. How are you ? Are you well ! Great. It work C.B.N'));

File: pos_tagging.py
import json

class posTagging :
    def __init__(self):
        try:
            self.treebank_dic = json.load(open('treebank_tagged_words', 'r'))
            self.wordnet_dic = json.load(open('wordnet_dictionary', 'r'))
            self.pos = json.load(open('pos.json','r'))
        except FileNotFoundError as e:
            print(f"Error: {e}")
            raise
        self.sentences = []
        self.words_types_list = []

    def parse(self, text):
        text = text.lower()
        current_sentence = ''

        for word in text.split():
            current_sentence += word + ' '
            if word[-1] in ('.', '!', '?'):
                self.sentences.append(current_sentence.strip())
                current_sentence = ''

        if current_sentence :
            self.sentences.append(current_sentence.strip())

        for sentence in self.sentences:
            for word in sentence.split():
                self.words_types_list.append(self.getWordType(word))

        return self.words_types_list

    def getWordType(self, word):
        types = set()
        if word[-1] in ('.', ',', ':', ';', '!', '?'):
            word = word[:-1]
            
        if '-' in word :
            types.add('Adjective')
            
        if word in self.treebank_dic:
            tag = self.treebank_dic[word]
            types.add(self.pos["TREEBANK_POS"].get(tag))
           
        elif word in self.wordnet_dic:
            pos_list = self.wordnet_dic[word]
            for pos in pos_list : 
                types.add(self.pos["WORDNET_POS"].get(pos))
                
        return {'word': word, 'type': list(types)}

tagger = posTagging()
print(tagger.parse('Either a boy or girl could sing.'))